app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt

```diff
@@ -45,23 +46,32 @@ class PlantAdapter : ListAdapter<Plant, PlantAdapter.ViewHolder>(PlantDiffCallba
                 LayoutInflater.from(parent.context), parent, false))
     }
 
-    private fun createOnClickListener(plantId: String): View.OnClickListener {
-        return View.OnClickListener {
-            val direction = PlantListFragmentDirections.ActionPlantListFragmentToPlantDetailFragment(plantId)
-            it.findNavController().navigate(direction)
+    private fun createOnClickListener(): OnPlantItemClickListener {
+        return object : OnPlantItemClickListener {
+            override fun onPlantItemClick(rootView: View, plant: Plant) {
+                val binding = DataBindingUtil.getBinding<ListItemPlantBinding>(rootView)
+                val navigatorExtras = FragmentNavigatorExtras(binding!!.plantItemImage to plant.plantId)
```

Comment:
My view here that using `!!` here is really unsafe idea.

I'd like to do:

```
DataBindingUtil.getBinding<ListItemPlantBinding>(rootView)?.let{
    // blablaba
} ?: run {
   // Error-handling, like post a toast or snackbar events through LiveData.
}
```
The reason is that the click event is meanless when binding is really for some reasons `null` and error-handlings give fallback.

Reply:
i did wrote the .let{} but it becomes too many lines just to generate a navigatorExtras. so i decide to let it be single line to be more concise and no reason it will be null.

do you have any suggestion how to use `ViewCompat.getTransitionName(view)`? the kotlin Pair expects a non-null String, but ViewCompats returns a nullable causing an error.